std/hash_map.zig

```diff
@@ -236,6 +242,11 @@ pub fn HashMap(comptime K: type, comptime V: type, comptime hash: fn (key: K) u3
             return null;
         }
 
+        /// Calls remove(), asserts that a kv pair is removed, and discards it.
+        pub fn removeAssertDiscard(hm: *Self, key: K) void {
+            assert(hm.remove(key) != null);
```

Comment:
Does it really make sense to `asset` here rather than e.g. panic?

Reply:
this method is a utility method that does exactly what the name implies. You can always call `remove()` instead and do your own check and panic if that makes sense for you, which would mean that the conditional code stays in the build even in release-fast mode.

My code (which is [here](https://github.com/thejoshwolfe/legend-of-swarkland/blob/4ae34758a2066745e1edc769e7034c3e631363f5/src/server/game_engine.zig#L237-L242) if you're curious) is programmed to always avoid key collisions, and i want to check that assumption during develpment, but it *should* be impossible to trip those asserts if everything is programmed correctly. That's exactly the case for assert rather than conditional panic.